centml/sdk/api.py

```diff
@@ -44,25 +45,44 @@ def get_compute(id):
         return api.get_compute_deployment_deployments_compute_deployment_id_get(id)
 
 
-def create_inference(name, image, port, hw_id, health, min_replicas, max_replicas, username, password, env):
+def create_inference(
+    name, image, port, is_private, hw_to_id_map, health, min_replicas, max_replicas, env, command, command_args, timeout
+):
+    if is_private:
+        triplet = client_certs.generate_ca_client_triplet(name)
+        # Handle automatic download of client private secrets
+        client_certs.save_pem_file(name, triplet.client_private_key, triplet.client_certificate)
     with get_api() as api:
         req = platform_api_client.CreateInferenceDeploymentRequest(
             name=name,
             image_url=image,
-            hardware_instance_id=hw_id,
-            env_vars={k: v for (k, v) in env},
-            secrets=(
-                platform_api_client.AuthSecret(username=username, password=password) if username and password else None
-            ),
             port=port,
+            hardware_instance_id=hw_to_id_map,
+            healthcheck=health,
             min_replicas=min_replicas,
             max_replicas=max_replicas,
-            timeout=0,
-            healthcheck=health,
+            env_vars=dict(env) if dict(env) else None,
+            command=[command] if command else None,
+            command_args=(list(command_args) if command and len(list(command_args)) > 0 else None),
+            timeout=timeout,
+            endpoint_certificate_authority=triplet.certificate_authority if is_private and triplet else None,
```

Comment:
nitpick: here you only need to check `if triplet`

Reply:
triplet is only created if is_private is true

Reply:
yeah removed the extra check